arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/raise/Builders.kt

```diff
@@ -84,6 +85,11 @@ public class ResultRaise(private val raise: Raise<Throwable>) : Raise<Throwable>
   @JvmName("bindAllResult")
   public fun <A> Iterable<Result<A>>.bindAll(): List<A> =
     map { it.bind() }
+
+  @RaiseDSL
+  @JvmName("bindAllResult")
```

Comment:
We may have a problem with this name clashing in the Arrow 2.0 branch, where `NonEmptyList` is defined as a value class.

Reply:
Not sure how it would differ when it being a `value class` 🤔 From Kotlin AST point-of-view it's a different type, and from JVM binary it has a different name.